src/ctapipe/image/modifications.py

```diff
@@ -100,6 +111,205 @@ def _smear_psf_randomly(
+            camera which are useful in calibration. Also required for
+            configuring the TelescopeParameter traitlets.
+        config: traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            This is mutually exclusive with passing a ``parent``.
+        parent: ctapipe.core.Component or ctapipe.core.Tool
+            Parent of this component in the configuration hierarchy,
+            this is mutually exclusive with passing ``config``
+
+        """
+
+        super().__init__(subarray=subarray, config=config, parent=parent, **kwargs)
+        self.rng = np.random.default_rng(self.rng_seed)
+
+        self.event_type_filter = NoiseEventTypeFilter(parent=self)
+
+        # Read in the waveforms in the NSB-only file. Store in a dictionary
+        # with one key per telescope, containing an array [n_events, n_gains,
+        # n_pixels, n_samples]
+        nsb_database = defaultdict(list)  # [nevents, ngains, npixels, nsamples]
+        with EventSource(
+            input_url=self.nsb_file, skip_calibration_events=False
+        ) as source:
+            for event in source:
+                if not self.event_type_filter(event):
+                    continue
+                for tel_id in event.trigger.tels_with_trigger:
+                    nsb_database[tel_id].append(event.r1.tel[tel_id].waveform)
+        nsb_database = {
+            tel_id: np.stack(waveforms) for tel_id, waveforms in nsb_database.items()
+        }
+
+        # Check that we have enough NSB-only events for all telescopes. We
+        # require that the number of NSB events for any telescope is at
+        # least two times the number of waveforms (=nsb_level) that we will add
+        # up. This is to avoid excessive correlation among the waveforms.
+        stats_ok = True
+        for tel_id in nsb_database:
+            nevents = nsb_database[tel_id].shape[0]
+            if nevents >= 2 * self.nsb_level:
+                continue
+            self.log.error(
+                f"Not enough NSB events available for tel_"
+                f"id {tel_id}. "
+                f"For nsb_level = {self.nsb_level}, at least "
+                f"{2*self.nsb_level} events are needed ({nevents} "
+                f"were found)."
+            )
+            stats_ok = False
+
+        if not stats_ok:
+            raise ValueError("Please use an input nsb_file with more events!")
+
+        # Now shift the waveforms so that they have mean=0 and do not introduce
+        # any bias (just fluctuations). For each telescope and gain we average
+        # all pixels
+        for tel_id in nsb_database:
+            hg_mean = np.mean(nsb_database[tel_id][:, 0, :, :])  # HG
+            lg_mean = np.mean(nsb_database[tel_id][:, 1, :, :])  # LG
+            nsb_database[tel_id][:, 0, :, :] -= hg_mean
+            nsb_database[tel_id][:, 1, :, :] -= lg_mean
```

Comment:
```suggestion
            for channel in nsb_database[tel_id].shape[0]:
                mean = np.mean(nsb_database[tel_id][:, channel, :, :])
                nsb_database[tel_id][:, channel, :, :] -= mean
```
otherwise it won't work for single-gain cameras.